app/scripts/metamask-controller.js

```diff
@@ -7474,4 +7485,83 @@ export default class MetamaskController extends EventEmitter {
   #getGlobalNetworkClientId() {
     return this.networkController.state.selectedNetworkClientId;
   }
+
+  #initControllers({ initObjects, initState }) {
```

Comment:
These two functions could go in a `app/scripts/controller-init/utils.ts` and we pass the init request as an argument.